trieste/acquisition/multi_objective/partition.py

```diff
@@ -64,46 +89,83 @@ def __post_init__(self) -> None:
         tf.debugging.assert_shapes([(self.lower_idx, ["N", "D"]), (self.upper_idx, ["N", "D"])])
 
 
-class Pareto:
+class BoundIndexPartition(NonDominatedPartition):
```

Comment:
If this class is never supposed to be used directly, and the only difference between different BoundedIndexPartition subclasses is initialisation (i.e. there's no abstract method that they share), then you can add this:

```python
    def __new__(cls, *args, **kwargs):
        if cls is BoundIndexPartition:
            raise TypeError("BoundIndexPartition may not be instantiated directly")
        return object.__new__(cls, *args, **kwargs)
```

Reply:
Not sure if properly understood the whole workflow of this method, but now have put `Any` for args and return to for mypy, please check if this proper.